src/Egulias/EmailValidator/Parser/Parser.php

```diff
@@ -36,14 +36,18 @@ protected function validateQuotedPair()
     }
 
     /**
-     * @return string the the comment
-     * @throws \InvalidArgumentException
+     * @param int $openedParenthesis
+     * @return string the comment
      */
-    protected function parseComments()
+    protected function parseComments(&$openedParenthesis = 0)
```

Comment:
I was thinking of it, but it looked not obvious for method parseComments to return number of parenthesis.